tests/manage/pv_services/pvc_resize/test_pvc_expansion_when_full.py

```diff
@@ -0,0 +1,265 @@
+                        near_full_msg = (
+                            f"PVC {pvc_obj.name} is nearing full. Data "
+                            f"deletion or PVC expansion is required."
+                        )
+                        check_alert_list(
+                            label='PersistentVolumeUsageNearFull',
+                            msg=near_full_msg, alerts=alerts_pvc,
+                            states=['firing'], severity='warning'
+                        )
+                        pvc_obj.near_full_alert = True
+                    except AssertionError:
+                        log.info(
+                            f"'PersistentVolumeUsageNearFull' alert not "
+                            f"started firing for PVC {pvc_obj.name}"
+                        )
+
+                # Verify 'PersistentVolumeUsageCritical' alert is firing
+                if not getattr(pvc_obj, 'critical_alert', False):
+                    try:
+                        log.info(
+                            f"Checking 'PersistentVolumeUsageCritical' alert "
+                            f"for PVC {pvc_obj.name}"
+                        )
+                        critical_msg = (
+                            f"PVC {pvc_obj.name} is critically full. Data "
+                            f"deletion or PVC expansion is required."
+                        )
+                        check_alert_list(
+                            label='PersistentVolumeUsageCritical',
+                            msg=critical_msg, alerts=alerts_pvc,
+                            states=['firing'], severity='error'
+                        )
+                        pvc_obj.critical_alert = True
+                    except AssertionError:
+                        log.info(
+                            f"'PersistentVolumeUsageCritical' alert not "
+                            f"started firing for PVC {pvc_obj.name}"
+                        )
+
+            # Collect list of PVCs for which alerts are not firing
+            not_near_full_pvc = [
+                pvc_ob.name for pvc_ob in self.pvcs if not getattr(
+                    pvc_ob, 'near_full_alert', False
+                )
+            ]
+            not_critical_pvc = [
+                pvc_ob.name for pvc_ob in self.pvcs if not getattr(
+                    pvc_ob, 'critical_alert', False
+                )
+            ]
+
+            if (not not_near_full_pvc) and (not not_critical_pvc):
+                log.info(
+                    "'PersistentVolumeUsageNearFull' and "
+                    "'PersistentVolumeUsageCritical' alerts are firing "
+                    "for all PVCs."
+                )
+                break
+
+        log.info("Expanding PVCs.")
+        for pvc_obj in self.pvcs:
+            log.info(
+                f"Expanding size of PVC {pvc_obj.name} to "
+                f"{pvc_size_expanded}Gi"
+            )
+            pvc_obj.resize_pvc(pvc_size_expanded, True)
```

Comment:
are we expecting this to pass?

Reply:
Yes. Verification will be done in resize_pvc function itself.

Reply:
But isn't the disk full while expanding?

Reply:
The cluster storage will have space available. Only this particular volume is full. So expansion will work as normal.

Reply:
got you!